Code/Framework/AzCore/AzCore/Date/DateFormat.h

```diff
@@ -0,0 +1,456 @@
+        {
+            return false;
+        }
+
+        // Update the output parameters and return success
+        yearMonthDay = currentYmd;
+        hoursMinutesSeconds = currentTimeOfDay;
+        return true;
+    }
+}
+
+namespace AZ::Date
+{
+    //! Contains methods to convert a chrono UTC clock time point into
+    //! ISO8601 timestamp per https://www.iso.org/iso-8601-date-and-time-format.html
+    using Iso8601TimestampString = AZStd::fixed_string<64>;
+
+    //! Date Formats
+    // formats an extended format ISO 8601 timestamp date
+    // Returns true if utcTime is valid
+    template<class Duration>
+    bool GetIso8601ExtendedFormatDate(Iso8601TimestampString& utcTimestamp, const AZStd::chrono::utc_time<Duration>& utcTime)
+    {
+        using DurationType = typename AZStd::remove_cvref_t<decltype(utcTime)>::duration;
+
+        AZStd::chrono::year_month_day currentYmd;
+        [[maybe_unused]] AZStd::chrono::hh_mm_ss<DurationType> currentTimeOfDay;
+        if (!Internal::GetYearMonthDayWithHoursMinutesSecondsFromUTCTime(currentYmd, currentTimeOfDay, utcTime))
+        {
+            return false;
+        }
+
+        // Print the current utc timestamp
+        // using the format specified at https://www.w3.org/TR/NOTE-datetime
+        utcTimestamp = Iso8601TimestampString::format(
+            "%.4i-%.2u-%.2u",
+            int{ currentYmd.year() },
+            unsigned{ currentYmd.month() },
+            unsigned{ currentYmd.day() });
+
+            return true;
+    }
+
+    // formats an basic format ISO 8601 timestamp date
+    // This is suitable for filenames
+    // It should be avoid in plain text per the ISO 8601 standard
+    // Returns true if utcTime is valid
+    template<class Duration>
+    bool GetIso8601BasicFormatDate(Iso8601TimestampString& utcTimestamp, const AZStd::chrono::utc_time<Duration>& utcTime)
+    {
+        using DurationType = typename AZStd::remove_cvref_t<decltype(utcTime)>::duration;
+
+        AZStd::chrono::year_month_day currentYmd;
+        [[maybe_unused]] AZStd::chrono::hh_mm_ss<DurationType> currentTimeOfDay;
+        if (!Internal::GetYearMonthDayWithHoursMinutesSecondsFromUTCTime(currentYmd, currentTimeOfDay, utcTime))
+        {
+            return false;
+        }
+
+        // Print the current utc timestamp
+        // using the format specified at https://www.w3.org/TR/NOTE-datetime
+        utcTimestamp = Iso8601TimestampString::format(
+            "%.4i%.2u%.2u",
+            int{ currentYmd.year() },
+            unsigned{ currentYmd.month() },
+            unsigned{ currentYmd.day() });
+
+            return true;
+    }
+
+    //! Time formats
+    // formats an extended format ISO 8601 timestamp time to seconds
+    // This format is not suitable for filenames as it contains the colon character, which isn't allowed
+    // for windows filesystems(NTFS)
+    // Returns true if utcTime is valid
+    // Use the basic format for filesystem safe timestamp
```

Comment:
question: Do we explicitly not what these comments to be picked up by doxygen? Otherwise we should likely update them all to be `//!` style (repeated across the other functions too)

Reply:
Yeah I will add the `!`